vm/isolated/isolated.go

```diff
@@ -324,14 +363,50 @@ func (inst *instance) Run(timeout time.Duration, stop <-chan bool, command strin
 	if inst.debug {
 		tee = os.Stdout
 	}
-	merger := vmimpl.NewOutputMerger(tee)
-	merger.Add("dmesg", dmesg)
-	merger.Add("ssh", rpipe)
+	inst.merger = vmimpl.NewOutputMerger(tee)
+	inst.merger.Add("dmesg", dmesg)
+	inst.merger.Add("ssh", rpipe)
 
-	return vmimpl.Multiplex(cmd, merger, dmesg, timeout, stop, inst.closed, inst.debug)
+	return vmimpl.Multiplex(cmd, inst.merger, dmesg, timeout, stop, inst.closed, inst.debug, false)
 }
 
+func (inst *instance) readPstoreContents() ([]byte, error) {
+	log.Logf(0, "reading pstore contents")
+	dir, err := ioutil.TempDir("", "pstore")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+	args := append(vmimpl.SCPArgs(inst.debug, inst.sshKey, inst.targetPort),
+		inst.sshUser+"@"+inst.targetAddr+":"+pstoreConsoleFile, dir)
+	cmd := osutil.Command("scp", args...)
+	if err := cmd.Run(); err != nil {
+		return nil, err
+	}
+	inst.ssh(fmt.Sprintf("rm %v", pstoreConsoleFile))
+	c, err := ioutil.ReadFile(filepath.Join(dir, "console-ramoops-0"))
+	if err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+// TODO: Diagnose() now does more, so should it return an error as well?
```

Comment:
I am not sure what caller will be able to do with the error besides what you already do here (append to output).